packages/embark-accounts-manager/src/index.ts

```diff
@@ -0,0 +1,117 @@
+import async from "async";
+import {Embark, Events, Logger} /* supplied by @types/embark in packages/embark-typings */ from "embark";
+import {__} from "embark-i18n";
+import {AccountParser, dappPath} from "embark-utils";
+import Web3 from "web3";
+
+import fundAccount from "./fundAccount";
+
+export default class AccountsManager {
+  private readonly logger: Logger;
+  private readonly events: Events;
+  private accounts: any[];
+  private nodeAccounts: string[];
+  private web3: any;
+  private ready: boolean;
+
+  constructor(private readonly embark: Embark, _options: any) {
+    this.logger = embark.logger;
+    this.events = embark.events;
+    this.accounts = [];
+    this.nodeAccounts = [];
+    this.ready = false;
+
+    this.events.setCommandHandler("accounts-manager:onReady", (cb) => {
```

Comment:
we shouldn't need this ready stuff, what is the reason for it? if it's due to race conditions there are betters way, see for e.g https://github.com/embark-framework/embark/blob/refactor_5_0_0/packages/embark/src/cmd/cmd_controller.js#L194 and/or https://github.com/embark-framework/embark/blob/refactor_5_0_0/packages/embark/src/cmd/cmd_controller.js#L236